### PR TITLE
MINOR: Use FetchRequest v8 and ListOffsetRequest v3 in ReplicaFetcherThread

### PR DESCRIPTION
**Pull request by ijuma**
If inter.broker.protocol.version is 2.0-IV1 or newer. Also fixed ListOffsetRequest
so that v2 is used, if applicable.

Added a unit test that verifies that we use the latest version of the various
requests by default. Included a few minor tweaks to make testing easier.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/QuotaFactory.scala [hunk 1]
```diff
@@ -37,7 +37,8 @@ object QuotaFactory extends Logging {
 
   object UnboundedQuota extends ReplicaQuota {
     override def isThrottled(topicPartition: TopicPartition): Boolean = false
-    override def isQuotaExceeded(): Boolean = false
+    override def isQuotaExceeded: Boolean = false
```
Comment (rajinisivaram) [inline core/src/main/scala/kafka/server/QuotaFactory.scala:40]: Looks like the build failed because the usages of `isQuotaExceeded()` weren't updated?

Comment (ijuma) [reply core/src/main/scala/kafka/server/QuotaFactory.scala:40]: Oops, not sure how that happened. Will fix.

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM

**Comment by ijuma** [comment]
Merging to trunk and 2.0.

**Pull request merged**